server: TestAdminAPIJobs failed

server.TestAdminAPIJobs [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3359756&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3359756&tab=artifacts#/) on master @ [ab1fc343c9a1140191f96353995258e609a84d02](https://github.com/cockroachdb/cockroach/commits/ab1fc343c9a1140191f96353995258e609a84d02):


```
=== RUN   TestAdminAPIJobs
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAdminAPIJobs728054158
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestAdminAPIJobs
    admin_test.go:1578: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAdminAPIJobs728054158
--- FAIL: TestAdminAPIJobs (0.41s)
=== RUN   TestAdminAPIJobs/isAdmin=true#01
    admin_test.go:1574: 0: expected job IDs [5 4 3 2 1 687774249346498561 687774249347645441 687774249236234241 687774249237577729 687774249435103233], but got [5 4 3 2 1 687774249435103233 687774249346498561 687774249347645441 687774249236234241 687774249237577729]
    --- FAIL: TestAdminAPIJobs/isAdmin=true#01 (0.04s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestAdminAPIJobs PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
benesch marked as alumn{us/a}; resolving to nvanbenschoten instead
```
</p>
</details>

/cc @cockroachdb/server nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdminAPIJobs.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/server/admin_test.go
+++ b/pkg/server/admin_test.go
@@ -1570,6 +1570,13 @@ func TestAdminAPIJobs(t *testing.T) {
 				expected = testCase.expectedIDsViaNonAdmin
 			}
 
+			sort.Slice(expected, func(i, j int) bool {
+				return expected[i] < expected[j]
+			})
+
+			sort.Slice(resIDs, func(i, j int) bool {
+				return resIDs[i] < resIDs[j]
+			})
 			if e, a := expected, resIDs; !reflect.DeepEqual(e, a) {
 				t.Errorf("%d: expected job IDs %v, but got %v", i, e, a)
 			}

